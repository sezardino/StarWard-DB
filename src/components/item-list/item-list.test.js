import React from 'react';
import {shallow} from 'enzyme';

import ItemList from './item-list';

describe('Component ItemList', () => {
	const data = [
		{id: '1', name: 'Test name'},
		{id: '2', name: 'Test name'},
	];
	const onListItemClick = jest.fn();
	const renderItem = (item) => item.name;
	const component = shallow(
		<ItemList data={data} renderItem={renderItem} onListItemClick={onListItemClick} />
	);

	it('Component rendered correctly', () => {
		expect(component.debug()).toMatchSnapshot();
	});

	describe('Component handlers', () => {
		it('onListItemClick has been called', () => {
			const item = component.find('.list-group-item:first-child');
			expect(onListItemClick).toBeCalledTimes(0);
			item.simulate('click');
			expect(onListItemClick).toBeCalledTimes(1);
			item.simulate('click');
			expect(onListItemClick).toBeCalledTimes(2);
		});
	});
});
