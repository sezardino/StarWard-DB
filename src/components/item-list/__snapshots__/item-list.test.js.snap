// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ItemList Component rendered correctly 1`] = `
"<ul className=\\"item-list list-group\\">
  <li className=\\"list-group-item\\" onClick={[Function: onClick]}>
    Test name
  </li>
  <li className=\\"list-group-item\\" onClick={[Function: onClick]}>
    Test name
  </li>
</ul>"
`;
